zerver/tests/test_subs.py

```diff
@@ -241,8 +241,13 @@ def test_api_fields(self) -> None:
         in `APIStreamDict` and `APISubscriptionDict`, respectively.
         """
         expected_fields = set(Stream.API_FIELDS) | {"stream_id", "is_archived"}
-        expected_fields -= {"id", "can_remove_subscribers_group_id", "deactivated"}
-        expected_fields |= {"can_remove_subscribers_group"}
+        expected_fields -= {
+            "id",
+            "can_remove_subscribers_group_id",
+            "can_administer_channel_group_id",
+            "deactivated",
+        }
+        expected_fields |= {"can_remove_subscribers_group", "can_administer_channel_group"}
```

Comment:
Similarly, this logic probably wants to use the `stream_permission_group_settings` data structures to compute the expected fields without needing every new setting to require editing this test.